### PR TITLE
[SPARK-23154][ML][DOC] Document backwards compatibility guarantees for ML persistence

### PR DESCRIPTION
**Pull request by jkbradley**
## What changes were proposed in this pull request?

Added documentation about what MLlib guarantees in terms of loading ML models and Pipelines from old Spark versions.  Discussed & confirmed on linked JIRA.

### CODE REVIEW

**Comment by jkbradley** [comment]
Here's what it looks like:

screen shot 2018-02-12 at 4 34 38 pm

**Comment by WeichenXu123** [comment]
LGTM.

File: docs/ml-pipeline.md [hunk 1]
```diff
@@ -188,9 +188,36 @@ Parameters belong to specific instances of `Estimator`s and `Transformer`s.
...
+Often times it is worth it to save a model or a pipeline to disk for later use. In Spark 1.6, a model import/export functionality was added to the Pipeline API.
+As of Spark 2.3, the DataFrame-based API in `spark.ml` and `pyspark.ml` has complete coverage.
+
+ML persistence works across Scala, Java and Python.  However, R currently uses a modified format,
+so models saved in R can only be loaded back in R; this should be fixed in the future and is
+tracked in [SPARK-15572](https://issues.apache.org/jira/browse/SPARK-15572).
+
+### Backwards compatibility for ML persistence
+
+In general, MLlib maintains backwards compatibility for ML persistence.  I.e., if you save an ML
```
Comment (viirya) [inline docs/ml-pipeline.md]: an ML -> a ML

**Review by viirya — COMMENTED** [review]
One minor comment. LGTM

File: docs/ml-pipeline.md (hunk 1 above)
Comment (jkbradley) [reply docs/ml-pipeline.md]: Yep, what @HyukjinKwon said.  Gotta love the irregularities of the language!

**Comment by jkbradley** [comment]
Thanks for reviewing!  I'll merge this with master and branch-2.3

**Pull request closed without merging**